docker-shim/Dockerfile

```diff
@@ -0,0 +1,19 @@
+# Determinate Nix Installer: Docker Shim
+#
+# This empty image exists to lean on Docker as a process supervisor when
+# systemd isn't available. Specifically intended for self-hosted GitHub
+# Actions runners using Docker-in-Docker.
+#
+# See: https://github.com/DeterminateSystems/nix-installer-action 
+
+FROM scratch
+
+ENTRYPOINT [ "/nix/var/nix/profiles/default/bin/nix-daemon"]
+CMD []
+
+HEALTHCHECK \
+    --interval=5m \
+    --timeout=3s \
+    CMD ["/nix/var/nix/profiles/default/bin/nix", "store", "ping", "--store", "daemon"]
+
+COPY ./Dockerfile /README.md
```

Comment:
...seems weird to be copying this Dockerfile into a `README.md` file in the root of the image...?

Reply:
It turns out if the Docker image is *actually* empty, a lot of Docker's tools break. So I figured the most friendly thing to do is to copy in the Dockerfile to give context about what the thing is. I didn't overthink it, just wanted to put something not entirely useless inside the container.

Reply:
Right, it just seems weird that you're copying in the _Dockerfile_ and not the, y'know, `README.md` in the same directory :laughing:

Reply:
The README is for us :)

Reply:
I think it would make more sense if the Dockerfile were copied to `/Dockerfile`, but I won't block on that.